lib/IDE/SwiftSourceDocInfo.cpp

```diff
@@ -312,8 +312,6 @@ std::pair<bool, ArgumentList *>
 NameMatcher::walkToArgumentListPre(ArgumentList *ArgList) {
   auto Labels = getCallArgLabelRanges(getSourceMgr(), ArgList,
                                       LabelRangeEndAt::BeforeElemStart);
-  tryResolve(Parent, ArgList->getStartLoc(), LabelRangeType::CallArg,
-             Labels.first, Labels.second);
```

Comment:
I believe this was trying to preserve the original logic for parens and tuples:

https://github.com/apple/swift/commit/78d8d09b995c2663f725f01e1ea4f99a1c7a6a62#diff-5fb3c14eb2d2fd312df17ef5b7e60ab238f2ef17a9944c3feede1373bc64d391L385-L389

Though it seems that would have previously been skipped for implicit argument lists, which therefore would therefore be skipped for operator argument lists.

It seems like it was originally added in https://github.com/apple/swift/pull/30964 to handle `callAsFunction` cases, is this still needed @nathawes?

Reply:
Yeah I think we might need to keep it– indexing uses the open paren as the base location of the call for callAsFunction (since there's no function name).

Reply:
Makes sense, we should probably restore the `!isImplicit` check in that case

Reply:
Good idea 👍

Reply:
I think it’s already covered by https://github.com/apple/swift/blob/main/test/refactoring/SyntacticRename/call-as-function.swift and a few others (which did fail before the latest commit).